File: go/EventService.go
package main

import (
	"io/ioutil"
	"log"
	"math/rand"
	"net/http"
	"os"
	"strconv"
	"time"
)

func main() {
	for true {
		time.Sleep(5000)
		orderId := rand.Intn(1000-1) + 1
		uri := "http://localhost:6001/order/" + strconv.Itoa(orderId)
		response, err := http.Get(uri)

		if err != nil {
			os.Exit(1)
		}

		result, err := ioutil.ReadAll(response.Body)
		if err != nil {
			log.Fatal(err)
		}

		log.Println("Order processed for order id " + strconv.Itoa(orderId))
		log.Println(result)
	}
}

File: service_invocation/python/EventService.py
import random
from time import sleep    
import requests
import logging
logging.basicConfig(level = logging.INFO)
    

while True:
    sleep(random.randrange(50, 5000) / 1000)
    orderId = random.randint(1, 1000)
    uri = 'http://localhost:6001/order/' + str(orderId)
    result = requests.get(uri)
    logging.info('Order processed for order id ' + str(orderId))
    logging.info(result)
    

